ssr-express/lib/AmpSsrMiddleware.js

```diff
@@ -97,10 +107,23 @@ class AmpSsrMiddleware {
         // This is a request for the canonical URL. Generate the AMP equivalent
         // in order to add it to the link rel tag.
         const linkRelAmpHtmlUrl = urlMapping.toAmpUrl(req.url);
-        ssr.transformHtml(body, {ampUrl: linkRelAmpHtmlUrl})
+
+        runtimeVersion()
+          .then(version => version)
+          .catch(err => {
+            console.error('Error retrieving ampRuntimeVersion: ', err);
+            return null;
+          })
+          .then(version => ssr.transformHtml(body, {
+            ampUrl: linkRelAmpHtmlUrl,
+            ampRuntimeVersion: version
+          }))
           .then(transformedBody => {
-            res.write(Buffer.from(transformedBody, 'utf-8'));
-            res.end();
+            res.status(200).send(transformedBody);
+          })
+          .catch(err => {
```

Comment:
Do multiple catch handlers work in a promise chain?

Reply:
They do!

Reply:
Will the promise chain stop after the first handler?

Reply:
The first `catch` handler is returning null, which will be passed to the `then` handler after it. So, the version will be null if retrieving the runtimeVersion fails.

If any error happens from then on, the second `catch` handler will be triggered.

Reply:
The last test on the Spec is testing this behaviour.